xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -432,7 +433,7 @@ def parse_optional_variable(
             match region_def:
                 case OptRegionDef():
                     return OptionalRegionVariable(variable_name, idx)
-                case VarRegionDef():
+                case VarRegionDef() | VarSingleBlockRegionDef():
```

Comment:
@superlopuh needed to fix this bug but was easy other than that

Reply:
Could alternatively make `VarSingleBlockRegionDef` inherit from `VarRegionDef`

Reply:
LGTM, would be good for @math-fehr to also approve this fix. I think the `|` is neater than inheritance, but I'm not sure if I'm missing some semantic nuance